### PR TITLE
(CONT-1123) - Add Puppet 8/Drop Puppet 6

### PR DESCRIPTION
**Pull request by Ramesh7**
## Summary
Updating puppet agent and puppet version dependency.

## Additional Context
Add any additional context about the problem here. 
- [x] Updating dependent module version to keep satellite_pe_tools updated 

## Related Issues (if any)
N/A

## Checklist
- [ ] 🟢 Spec tests.
- [ ] 🟢 Acceptance tests.
- [ ] Manually verified. (For example `puppet apply`)

### CODE REVIEW

**Comment by jordanbreen28** [comment]
PR needs `backwards-incompatible` label, we're removing support for Puppet 6 here.

File: spec/classes/init_spec.rb [hunk 1]
```diff
@@ -1,13 +1,28 @@
...
+    context "On #{os}" do
+      let(:facts) { facts }
 
-    include_examples 'RedHat 7'
+      context 'with defaults for all parameters' do
+        it { is_expected.to contain_class('satellite_pe_tools') }
+        it { is_expected.to contain_ini_subsetting('reports_satellite') }
+        it { is_expected.to contain_file('satellite_config_yaml') }
+        it { is_expected.to contain_exec('download_install_katello_cert_rpm') }
+        it { is_expected.to contain_file('/etc/puppetlabs/puppet/ssl/ca/katello-default-ca.crt') }
```
[the remark this answers is not in the document]
Comment (Ramesh7) [reply spec/classes/init_spec.rb:18]: While fixing test cases I thought we could add more here as many resources were untouched so added thinking that will be again same cycle of creating PR and getting reviewed so better to include into this only as there is no code change.

File: .rubocop_todo.yml [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+# The point is for the user to remove these configuration records
+# one by one as the offenses are removed from the code base.
+# Note that changes in the inspected code, or installation of new
+# versions of RuboCop, may require this file to be generated again.
+
+# Offense count: 1
+# This cop supports safe autocorrection (--autocorrect).
+Layout/ClosingHeredocIndentation:
+  Exclude:
+    - 'lib/puppet/parser/functions/parse_url.rb'
```
[the remark this answers is not in the document]
Comment (Ramesh7) [reply .rubocop_todo.yml]: Accommodated all auto correct warning

File: spec/classes/init_spec.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (jordanbreen28) [reply spec/classes/init_spec.rb:18]: yeah agreed. 
Just don't think it makes too much sense under a commit `Update agent and puppet version`, but i'll let you decide which is best.

File: lib/puppet/reports/satellite.rb [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # frozen_string_literal: true
 
-require __dir__ + '/../util/satellite.rb'
+require "#{__dir__}/../util/satellite.rb"
```
Comment (jordanbreen28) [inline lib/puppet/reports/satellite.rb:3]: this is likely to break the module, see https://github.com/puppetlabs/puppetlabs-registry/pull/287.
opt to use `require_relative`

File: .rubocop_todo.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (jordanbreen28) [reply .rubocop_todo.yml]: AH. All cops really should have been corrected in their own commit as per the other PRs.
Can you bundle them together in a seperate commit `(CONT-1123) - Rubocop fixes`

File: spec/classes/init_spec.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (Ramesh7) [reply spec/classes/init_spec.rb]: Created new story for adding test cases.

File: .rubocop_todo.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (Ramesh7) [reply .rubocop_todo.yml]: don!!

**Review by chelnak — APPROVED** [review]

**Pull request merged**